recipes/verilator/all/conanfile.py

```diff
@@ -1,204 +1,196 @@
-    description = "Verilator compiles synthesizable Verilog and Synthesis assertions into single- or multithreaded C++ or SystemC code"
-    topics = ("verilog", "hdl", "eda", "simulator", "hardware", "fpga")
+    topics = ("verilog", "hdl", "eda", "simulator", "hardware", "fpga", "pre-built")
 
+    package_type = "application"
     settings = "os", "arch", "compiler", "build_type"
 
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
-    def export_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
-            
-    @property
-    def _needs_old_bison(self):
-        return Version(self.version) < "4.100"
-
-    def build_requirements(self):
-        if self._settings_build.os == "Windows" and "CONAN_BASH_PATH" not in os.environ:
-            if self.settings.compiler == "Visual Studio":
-                self.build_requires("msys2/cci.latest")
-                self.build_requires("automake/1.16.4")
-            if self._needs_old_bison:
-                # don't upgrade to bison 3.7.0 or above, or it fails to build
-                # because of https://github.com/verilator/verilator/pull/2505
-                self.build_requires("winflexbison/2.5.22")
-            else:
-                self.build_requires("winflexbison/2.5.24")
-            self.build_requires("strawberryperl/5.30.0.1")
-        else:
-            self.build_requires("flex/2.6.4")
-            if self._needs_old_bison:
-                # don't upgrade to bison 3.7.0 or above, or it fails to build
-                # because of https://github.com/verilator/verilator/pull/2505
-                self.build_requires("bison/3.5.3")
-            else:
-                self.build_requires("bison/3.7.6")
-        if Version(self.version) >= "4.224":
-            self.build_requires("autoconf/2.71")
-
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
         if self.settings.os == "Windows":
-            self.requires("strawberryperl/5.30.0.1")
-        if self.settings.compiler == "Visual Studio":
-            self.requires("dirent/1.23.2", private=True)
+            self.requires("strawberryperl/5.32.1.1", visible=False)
+            if is_msvc(self):
+                self.requires("dirent/1.24", visible=False)
+        # Conan v1 does not handle flex being both a build and a host requirement correctly.
+        # It gets incorrectly marked as a dependency in CMake but a CMake module/config is not generated.
+        if conan_version >= 2:
+            if self.settings.os == "Windows":
+                self.requires("winflexbison/2.5.25", visible=False)
+            else:
+                self.requires("flex/2.6.4", visible=False)
```

Comment:
Why is flex need now, but not before? Using ldd on linux I don't see linkage. Please, update the PR description with the changes related to this PR, it really hard to track and review without any information.

Reply:
My memory is very fuzzy, it's been almost half a year, sorry. I think it was due to flex being a run-time requirement and getting Conan v1 and v2 to propagate it correctly in the graph were a major pain. I really don't want to touch this part of the recipe again, if possible